### ISSUE TITLE
Demonstration DETR training code

### ISSUE THREAD

**Issue by matyasbohacek** [facebookresearch/detr#300]
Hi,

I wanted to ask whether the training code for the simplified demonstration DETR ([Demo Colab here](https://colab.research.google.com/github/facebookresearch/detr/blob/colab/notebooks/detr_demo.ipynb)) is available publicly? In this example, only a model pretrained on COCO is loaded and there is no reference whatsoever to the specifics of the training or the code itself.

I have been trying to implement this on my own using the torch's [`CocoDetection`](https://pytorch.org/docs/stable/torchvision/datasets.html#detection) dataset, but I have been running into many issues for multiple days, still not making much progress.

I would very much appreciate any help on this! Otherwise thanks for a great set of tools to experiment with the architecture.

**Comment by lhuasheng** [comment]
You can try this link.

https://www.kaggle.com/tanulsingh077/end-to-end-object-detection-with-transformers-detr

**Comment by matyasbohacek** [comment]
Thanks, this definitely helped! I managed to write my own script for it.

**Comment by Supersak80** [comment]
@alcinos In the forward pass of https://colab.research.google.com/github/facebookresearch/detr/blob/colab/notebooks/detr_demo.ipynb:

why is 10% of the feature encoding (h) added to the positional encoding?

**Comment by alcinos** [comment]
It seems that the underlying issue has been resolved. See also #63 for more information.
However, I want to stress that it is really intended as a demo to understand the general data flow in the model and not something anyone should use in anything serious beyond tinkering. The "official" model is not that much more complicated, I encourage you to dig into it once you've understood how the minimal version works.

@Supersak80 the 0.1 rescaling that you are referring to is used to properly scale the contributions of the feature encoding with respect to the position encoding. It is not strictly required, but we have found helpful to apply it.

I believe all the questions have been answered, and as such I'm closing this. Feel free to reach out if you have further concerns.
